api/src/services/files.ts

```diff
@@ -83,11 +90,19 @@ export class FilesService extends ItemsService {
 		}
 
 		const fileExtension =
-			path.extname(payload.filename_download!) || (payload.type && '.' + extension(payload.type)) || '';
+			path.extname(payload.filename_download!) || (payload?.type && '.' + extension(payload.type)) || '';
+
+		if (!path.extname(payload.filename_download!) && payload?.type) {
```

Comment:
This checks if the `filename_download` doesn't exist, and there is a payload type, to upload the `filename_download` extension.